plugins/hls-semantic-tokens-plugin/src/Ide/Plugin/SemanticTokens/Mappings.hs

```diff
@@ -66,7 +77,7 @@ tyThingSemantic ty = case ty of
     | isTyVar vid -> Just TTypeVariable
     | isRecordSelector vid -> Just TRecordField
     | isClassOpId vid -> Just TClassMethod
-    | isFunVar vid -> Just TFunction
+    | isFunVar vid -> Just TFunction <> (nameInfixOperator $ getName vid)
```

Comment:
Since you're going to merge them all with `<>` later, perhaps `tyThingSemantic` should just return `[HsSemanticTokenType]`, since in general we might allocate several types? That would simplify some things, e.g. here you're implicitly duplicating the fact that `TClassMethod` is preferred over `TFunction` by putting the match for `isClassOpId` first. (So if you ever changed the preference in the Semigroup instance you would still get the old answer!).

So you could do something like:
```
AnId vid -> [TTypeVariable | isTyVar vid] ++ ... ++ [TFunction | isFunVar vid] ++ [TOperator | isSymId vid]
```
and then let the semigroup instance sort it all out. Less important for the other cases where we don't have overlapping token types, though.

Reply:
yes, and it is going to be more tricky when priorities is configurable. Since we need to make sure the behaviour consistant between information from hieAst and the one from typeThing.

Reply:
Well, that should work out okay with the model where we just collect (possibly many) token types and then pick the "best". We'll have to have a "best" function that is derived from the configuration, instead of just the hardcoded one based on the Semigroup instance, but that seems okay?

Reply:
I agree, we might want to switch to list or set eventually with priorities configurations. Since by then, we might need to add test to ensure the all the results coming from imported or local names would have the same set of token types(instead of the current strategy to have just the same token type).

Reply:
> test to ensure the all the results coming from imported or local names would have the same set of token types

You mean if I have 

```
module A where

foo = ...
```
```
module B where
import A

bar = foo
```
then `foo` should have the same tokens in both cases? That does seem like a nice property!

Reply:
Exactly